src/elexsolver/QuantileRegressionSolver.py

```diff
@@ -61,28 +61,43 @@ def get_loss_function(self, x, y, coefficients, weights):
         residual = y - y_hat
         return cp.sum(cp.multiply(weights, 0.5 * cp.abs(residual) + (self.tau.value - 0.5) * residual))
 
-    def get_regularizer(self, coefficients):
-        return cp.pnorm(coefficients, p=2) ** 2
-
-    def __solve(self, x, y, weights, lambda_, verbose):
+    def get_regularizer(self, coefficients, fit_intercept):
+        # if we are fitting an intercept in the model, then that coefficient should not be regularized.
+        # NOTE: assumes that if fit_intercept=True, that the intercept is in the first column
+        coefficients_to_regularize = coefficients
+        if fit_intercept:
+            coefficients_to_regularize = coefficients[1:]
```

Comment:
Can you put a check here that the first column is in fact the intercept column (e.g. check its all 1s) and throw error if its not?

Reply:
How about a warning in that case? I think there might be v weird reasons that you'd want to have a non 1s intercept column.

Reply:
what do we usually call that column? should we just check its the right name? Im more concerned of a column ordering mix-ups

Reply:
the solver unfortunately takes a numpy array and not a pandas array, so the columns no longer have names. I agree with you on the mixup issue, the thing that might save us in the elex-live-model specific case is that we force the intercept to be first here: https://github.com/washingtonpost/elex-live-model/blob/develop/src/elexmodel/handlers/data/Featurizer.py#L88

Reply:
fine -- lets do a warning rn if not all 1's